copy/copy.go

```diff
@@ -451,38 +451,51 @@ func (c *copier) copyMultipleImages(ctx context.Context, policyContext *signatur
 		return nil, "", errors.Wrapf(err, "Error updating manifest list")
 	}
 
-	// Perform the list conversion.
-	if selectedListType != list.MIMEType() {
-		list, err = list.ConvertToMIMEType(selectedListType)
+	// Iterate through supported list types, preferred format first.
+	c.Printf("Writing manifest list to image destination\n")
+	var errs []string
+	for _, thisListType := range append([]string{selectedListType}, otherManifestMIMETypeCandidates...) {
+		logrus.Debugf("Trying to use manifest list type %s…", thisListType)
+
+		// Perform the list conversion, if we need one.
+		if thisListType != list.MIMEType() {
+			list, err = list.ConvertToMIMEType(thisListType)
```

Comment:
Yeah, changing that so that `list` (now `updatedList`, because it's where the instances get updated) and `manifestList` are only modified if the remote accepts the converted version of the list.